### ISSUE TITLE
Give simple synchronized/blocked version of for each function to iterate the reader

### ISSUE THREAD

**Issue by defia** [gocarina/gocsv#56]
UnmarshalToCallback first create a channel , start a goroutine to fill the channel, then a loop to read from the channel, and call callback function to handle the object.
I think it's users choice to decide whether to handle the whole process in synchronize or async.

So just give us a function like filepath.Walk, so we can read the csv in streaming, and with a callback that  we can returns error to stop the iterate.

**Comment by CheerChen** [comment]
I agree with that. Just now I want to use UnmarshalToChan to read line by line and got into trouble using goroutine. It will be easier that reading the csv in synchronize way.
